docs/ruleset_functions/get_opaque_surface_type.md

```diff
@@ -0,0 +1,31 @@
+
+## get_opaque_surface_type
+
+Description: This function would determine whether it is a wall, ceiling or floor.  
+
+Inputs:
+  - **Surface**: The surface that needs to determine surface type.  
+
+Returns:
+- **opaque_surface_type**: The Opaque Surface Type [roof, heated slab-on-grade, unheated slab-on-grade, floor, above-grade wall, below-grade wall].  
+
+
+Logic:  
+
+- If surface tilt is more than or equal to 0 degree and less than 60 degrees, surface is classified as roof: ```if 0 <= surface.tilt < 60: surface_type = "ROOF"```  
+
+- Else if surface tile is more than 120 degrees and less than or equal to 180 degrees, surface is classified as floor: ```else if 120 < surface.tilt <= 180:```  
+
+  - Determine if surface is heated slab-on-grade: ```if ( surface.construction.has_radiant_heating ) AND ( surface.adjacent_to == "GROUND" ): surface_type = "HEATED SLAB-ON-GRADE"```  
+
+  - Else determine if surface is unheated slab-on-grade: ```else if surface.adjacent_to == "GROUND": surface_type = "UNHEATED SLAB-ON-GRADE"```  
+
```

Comment:
Based on Section 3, "slab-on-grade floor: that portion of a slab floor of the building envelope that is in contact
with the ground and that is either above grade or is less than or equal to 24 in. below the final elevation of the nearest exterior grade." We cannot differentiate slab-on-grade from slab-below-grade, so when baseline slab on grade insulation does not match the expected value, there should be a flag for manual review, to confirm the depth. Could include is as a note here so that we remember in the calling function.

Reply:
Note added.